### PR TITLE
Seller Experience: Add 'Simple' vs 'More Power' view

### PR DESCRIPTION
**Pull request by nelsonec87**
#### Changes proposed in this Pull Request

* Depends on [PR 60222](https://github.com/Automattic/wp-calypso/pull/60222)

* Added new view so the user can pick from a simpler (Payment Blocks) or more powerful (Woo) store.

#### Testing instructions

* Access `start/setup-site/intent?siteSlug=<SITE SLUG>&flags=seller-experience` (which enables the `seller-experience` feature flag)
* Click the "Start Selling" button
image
* Click "Continue"

image

* Check the new view
image

Related to #60155

### CODE REVIEW

File: client/signup/steps/site-options/index.tsx [hunk 1]
```diff
@@ -20,10 +21,32 @@ interface Props {
...
 	const { stepName, signupDependencies, goToNextStep } = props;
 	const { siteTitle, tagline } = signupDependencies;
 
+	const getStepText = ( stepName: string ) => {
+		switch ( stepName ) {
+			case 'store-options':
+				return {
+					headerText: translate( "First, let's give your store a name" ),
+					headerImage: intentImageUrl,
+					siteTitleLabel: translate( 'Store name' ),
```
Comment (a8ci18n) [inline client/signup/steps/site-options/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Store Name' )` *ES Score: 9*

File: client/signup/steps/site-options/index.tsx [hunk 2]
```diff
@@ -20,10 +21,32 @@ interface Props {
...
+					taglineExplanation: translate( 'In a few words, explain what your store is about.' ),
```
Comment (a8ci18n) [inline client/signup/steps/site-options/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **49** times:
`translate( 'In a few words, explain what your blog is about.' )` *ES Score: 7*

**Comment by kwight** [comment]
For now, let's have the "start with more" button link to https://wordpress.com/start/woocommerce-install/confirm?site=[domain]. The "WooCommerce" sidebar menu item of the new WooCommerce installation flow goes there. Unless @SaxonF knows of a better place?..

**Comment by SaxonF** [comment]
@kwight makes sense to me.

File: client/signup/steps/store-features/index.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+						{ translate(
+							'Ideal if you’re looking to accept donations or sell one or two products without needing to manage shipping.'
+						) }
+					</span>
+					<span className="store-features__powered-by">
+						<span className="store-features__image-wrapper">
+							<img src={ paymentBlocksImage } alt="Payment Blocks" />
+						</span>
+						<span>
+							{ translate( 'Powered by' ) } <button>Payment Blocks</button>
```
Comment (a8ci18n) [inline client/signup/steps/store-features/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'powered by' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/feature/seller-experience-add-simple-vs-power-view)

Comment (kwight) [inline client/signup/steps/store-features/index.tsx]: Strings shouldn't be broken up like this, even when we need markup in the middle – the markup is supported by `translate`, so this line would look more like https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/translations/seller-experience.js#L11

You can find more info about `translate` in its README: https://github.com/Automattic/wp-calypso/tree/trunk/packages/i18n-calypso

File: client/signup/steps/store-features/index.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+		dispatch( saveSignupStep( { stepName } ) );
+	}, [] );
+
+	const intents: SelectItem< any >[] = [
+		{
+			key: 'simple',
+			title: translate( 'Start Simple' ),
+			description: (
+				<>
+					<span className="store-features__requirements">{ translate( 'Free' ) }</span>
```
Comment (kwight) [inline client/signup/steps/store-features/index.tsx:44]: Spans for everything ends up being semantically meaningless; instead, default to `p`s when they're a block or line of their own. In general, try to make meaningful HTML5-compliant markup – I could see markdown like this (old-school `div`s are options too):

Screen Shot 2022-01-25 at 2 42 31 PM

File: client/signup/steps/store-features/index.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+			value: 'simple',
+			actionText: translate( 'Continue' ),
+		},
+		{
+			key: 'power',
+			title: translate( 'More Power' ),
+			description: (
+				<>
+					<span className="store-features__requirements">
+						{ translate( 'Requires {{link}}business plan{{/link}} for %(monthlyPrice)s/month', {
```
Comment (kwight) [inline client/signup/steps/store-features/index.tsx]: We can't hard-code prices or currency – let's do [this](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/translations/seller-experience.js#L22) for now, and then come back to it if others feel strongly about actual numbers here.

File: client/signup/steps/store-features/index.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+								link: <button />,
+							},
+						} ) }
+					</span>
+					<span>
+						{ translate(
+							'If you have multiple products or require extensive order and shipping management than this might suit your needs better.'
+						) }
+					</span>
+					<span className="store-features__powered-by">
```
Comment (kwight) [inline client/signup/steps/store-features/index.tsx]: Are two wrapping spans necessary here?

File: client/signup/steps/store-features/index.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+				</>
+			),
+			icon: truck,
+			value: 'power',
+			actionText: translate( 'Upgrade' ),
+		},
+	];
+
+	const onSelect = () => {
+		throw new Error( 'Not yet implemented' );
```
Comment (kwight) [inline client/signup/steps/store-features/index.tsx]: Nice. :)

File: client/signup/steps/store-features/index.tsx (hunk 4 above)
Comment (nelsonec87) [reply client/signup/steps/store-features/index.tsx:44]: I totally agree :) 
In fact, I tried to do it "right" the first time, but the whole description is a `p` in the wrapper component:

image


So I went with the `span`s for now. I'll look at the wrapper component to see what can be done without breaking other views.

File: client/signup/steps/store-features/index.tsx (hunk 3 above)
Comment (nelsonec87) [reply client/signup/steps/store-features/index.tsx]: Awesome! Your preemptive translations file is gonna be super helpful as well, thank you.

File: client/signup/steps/store-features/index.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+	signupDependencies: any;
+	stepName: string;
+	initialContext: any;
+}
+
+export default function StoreFeaturesStep( props: Props ): React.ReactNode {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const headerText = translate( 'Setup your store' );
+	const subHeaderText = translate( "Let's create a website that suits your needs" );
```
Comment (a8ci18n) [inline client/signup/steps/store-features/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **6** times:
`translate( 'Let’s create a website that suits your needs.' )` *ES Score: 6*

File: client/signup/steps/store-features/index.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+	// Only do following things when mounted
+	React.useEffect( () => {
+		dispatch( saveSignupStep( { stepName } ) );
+	}, [] );
+
+	const intents: SelectItem< StoreFeatureSet >[] = [
+		{
+			key: 'simple',
+			title: translate( 'Start Simple' ),
```
Comment (a8ci18n) [inline client/signup/steps/store-features/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **6** times:
`translate( 'Start simple' )` *ES Score: 10*

File: client/signup/steps/store-features/index.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+	signupDependencies: any;
+	stepName: string;
+	initialContext: any;
+}
+
+export default function StoreFeaturesStep( props: Props ): React.ReactNode {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const headerText = translate( 'Setup your store' );
+	const subHeaderText = translate( "Let's create a website that suits your needs." );
```
Comment (a8ci18n) [inline client/signup/steps/store-features/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **6** times:
`translate( 'Let’s create a website that suits your needs.' )` *ES Score: 6*

File: client/signup/steps/store-features/index.tsx (hunk 4 above)
Comment (nelsonec87) [reply client/signup/steps/store-features/index.tsx:44]: One idea to keep the markup meaningful is to replace the `<p>` from `SelectItems`'s markup for a generic `<div>` .

`div`s allow us to have nested `<p>` and other tags as well.

We could then add a `<p>` to the existing child components - keeping their meaning - and only 4 files would be affected by this change.

**Comment by nelsonec87** [comment]
> For now, let's have the "start with more" button link to [https://wordpress.com/start/woocommerce-install/confirm?site=[domain]](https://wordpress.com/start/woocommerce-install/confirm?site=%5Bdomain%5D). The "WooCommerce" sidebar menu item of the new WooCommerce installation flow goes there. Unless @SaxonF knows of a better place?..

Using https://wordpress.com/start/woocommerce-install?site=[domain] so the router figures out the path for Woo first step, as discussed.

File: client/signup/select-items/index.tsx [hunk 11]
```diff
@@ -30,7 +30,7 @@ function SelectItems< T >( { className, items, onSelect }: Props< T > ): React.R
 					<div className="select-items__item-info-wrapper">
 						<div className="select-items__item-info">
 							<h2 className="select-items__item-title">{ preventWidows( title ) }</h2>
-							<p className="select-items__item-description">{ preventWidows( description ) }</p>
+							<div className="select-items__item-description">{ preventWidows( description ) }</div>
```
Comment (kwight) [inline client/signup/select-items/index.tsx:33]: This is being done to allow more structured HTML in the description, like our more verbose need here in this step. Previous instances of `SelectItems` now have a `div` around them (not the best), but it seemed a better option than a whole new component to allow decent markup for our use-case. Another option is to accept a component for the "description" area. ¯\_(ツ)_/¯ We went with what seemed the best of a few not-great options.

File: client/signup/steps/store-features/index.tsx [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+	isReskinned: boolean;
+	signupDependencies: any;
+	stepName: string;
+	initialContext: any;
+}
+
+export default function StoreFeaturesStep( props: Props ): React.ReactNode {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const headerText = translate( 'Setup your store' );
```
Comment (kwight) [inline client/signup/steps/store-features/index.tsx]:
```suggestion
	const headerText = translate( 'Set up your store' );
```

**Comment by kwight** [comment]
@nelsonec87  After fixing the typo, go ahead and commit this so we can build on it. There are a few things we'll need to follow up on.

Button alignment: https://github.com/Automattic/wp-calypso/issues/60517
Woo icon quality: https://github.com/Automattic/wp-calypso/issues/60518

**Review by kwight — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7120048

Thank you @nelsonec87 for including a screenshot in the description! This is really helpful for our translators.

File: client/signup/select-items/index.tsx (hunk 11 above)
Comment (arthur791004) [reply client/signup/select-items/index.tsx:33]: This change increases the gap between the description and the action button on mobile. See p58i-c55-p2.

We should also update the style as follows:
```css
// client/signup/select-items/style.scss
.select-items__item {
  ...
  &-description {
    // margin-bottom: 0; <--- this should change to `p` element

    p {
      margin-bottom: 0;
    }
  }
  ...
}
```

See https://github.com/Automattic/wp-calypso/pull/60943

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7120048

Thank you @nelsonec87 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7120048).

**Pull request merged**